languages/ruby/exercises/concept/instance-variables/.meta/design.md

```diff
@@ -0,0 +1,38 @@
+## Goal
+
+The goal of this exercise is to teach the student the basics of the concept instance variables and "nil" in Ruby.
+
+## Learning objectives
+
+- Know that objects can store state using instance variables
+- Know that instance variables should be treated as private and to use getter and setter methods
+- Know that `nil` is a Ruby object used to represent "nothingness"
+- Know that instance variables are `nil` before they are assigned otherwise
+
+## Out of scope
+
+- Boolean values (True/False)
+- Truthy/Falsey-ness
+
+## Concepts
+
+- `instance-variables`
+- `nil`
```

Comment:
```suggestion
- Instance variables
- Nil
```

While `nil` is an object, we should not mark it as code here, as it is the concept instead.  If we want to mark these both with something, quotes, would be suitable or leave them plain.  (And I know at some point someone is going to want to quote mark `nil` as code.  Perhaps capitalize the first letter of each bullet point as well, to help avoid the reflex.

Reply:
these are the tokens that completing this exercise will unlock, which other exercises look for to see if students are able to begin that exercise.  It should stay this way AFAIK

Reply:
It is done this way in every other design.md file I have checked.  So "should stay this way" not positive, but currently it is the way it is.  The markup may be being used programmatically or may just be here because after the first one, the second one did so because: "This is the way it's always been done".  If in doubt, cargo cult. ;)

Reply:
In other ruby exercises' design.md, you are correct it has been done this way. But look at csharp's basics exercise, this section (and the prerequisites section) is used to list concept slugs, not proper programming concepts. Elixir, python have also followed suit with this.

There is the option of verbosely describing the concept here in this document (as csharp has done) but I haven't elected to do with elixir exercises because the concept slugs definitions are kept in a different file.

Reply:
Oh, I looked at all of them, did not think to only look at Ruby's.

```
at $(find -name design.md )
```

So didn't see Ruby's until much later down the list.